src/data/bucket/line_bucket.js

```diff
@@ -436,7 +444,7 @@ class LineBucket implements Bucket {
             // approximate angle from cosine
             const approxAngle = 2 * Math.sqrt(2 - 2 * cosHalfAngle);
 
-            const isSharpCorner = cosHalfAngle < COS_HALF_SHARP_CORNER && prevVertex && nextVertex;
+            const isSharpCorner = cosHalfAngle < COS_HALF_SHARP_CORNER && prevVertex && nextVertex && adjustSharpCorners;
```

Comment:
nit: Might be good to rename `isSharpCorner` to something else, because the corner is still sharp even if `adjustSharpCorners` is false. Maybe: `needsSharpCornerAdjustment`